tests/TestClasses/CharacterTransformer.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Spatie\Fractalistic\Test\TestClasses;
+
+use League\Fractal\Resource\Item;
+use League\Fractal\TransformerAbstract;
+
+final class CharacterTransformer extends TransformerAbstract
+{
+    protected array $availableIncludes = [
+        'book',
+        'author',
+    ];
+
+    public function transform(Character $character): array
+    {
+        $parentScope = last($this->getCurrentScope()->getParentScopes());
+        $data = [
+            'name' => $character->name,
+            'current_scope' => $this->getCurrentScope()->getScopeIdentifier(),
+            'parent_scope' => $parentScope,
+            'scope_identifier' => $this->getCurrentScope()->getIdentifier(),
+        ];
+
+        if ($parentScope === 'author') {
+            $data['called_by_author'] = 'indeed!';
+        }
+
+        if ($parentScope === 'books') {
+            $data['called_by_book'] = 'yes!';
+        }
```

Comment:
As you can see, we are returnning different fields based on the context that we are in.
In the `author` context we return, `called_by_author` field, and
In the `books` context we return, `called_by_book` field.
It seems this is the reult that #39 wanted to achiev.